Contentstack.Core/ContentstackClient.cs

```diff
@@ -146,21 +160,71 @@ public ContentstackClient(IOptions<ContentstackOptions> options)
         ///     ContentType contentType = stack.ContentType(&quot;contentType_name&quot;);
         /// </code>
         /// </example>
-        public ContentstackClient(string apiKey, string deliveryToken, string environment, string host = null, ContentstackRegion region = ContentstackRegion.US, string version = null) :
+        public ContentstackClient(string apiKey,
+                                string deliveryToken,
+                                string environment,
+                                string host = null,
+                                ContentstackRegion region = ContentstackRegion.US,
+                                bool retryOnError = true,
+                                string version = null,
+                                string proxyHost = null,
+                                int proxyPort = -1,
+                                ICredentials proxyCredentials = null) :
         this(new OptionsWrapper<ContentstackOptions>(new ContentstackOptions()
         {
             ApiKey = apiKey,
             DeliveryToken = deliveryToken,
             Environment = environment,
             Host = host,
             Region = region,
-            Version = version
+            Version = version,
+            RetryOnError = retryOnError,
+            ProxyHost = proxyHost,
+            ProxyPort = proxyPort,
+            ProxyCredentials = proxyCredentials
         }
         ))
         {
 
         }
 
+        protected void Initialize()
+        {
+            HttpClientHandler httpClientHandler = new HttpClientHandler
+            {
+                Proxy = _options.GetWebProxy()
+            };
+            _httpClient = new HttpClient(httpClientHandler);
+
+            _httpClient.DefaultRequestHeaders.Add(HeadersKey.XUserAgentHeader, $"{xUserAgent}");
+
+            if (!_httpClient.DefaultRequestHeaders.UserAgent.Any())
+            {
+                _httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("contentstack-dotnet", Version));
+                _httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("DotNet", Environment.Version.ToString()));
+            }
+
+            if (_options != null)
+            {
+                _httpClient.Timeout = _options.Timeout;
+                //httpClient.MaxResponseContentBufferSize = _options.MaxResponseContentBufferSize;
+                //LogManager = _options.DisableLogging ? LogManager.EmptyLogger : LogManager.GetLogManager(GetType());
+            }
+        }
+
+        protected void BuildPipeline()
```

Comment:
@AnkurContentstack Code Copied from Management SDK.

Reply:
Yes because same thing is implemented in management SDK